app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -249,10 +310,25 @@ fun MapScreen(
                   if (mapView.mapboxMap.cameraState.zoom > thresholdDisplayZoom) {
                     parkingViewModel.getParkingsInRect(currentBottomLeft, currentTopRight)
                   }
+                  // On zoom-out past the threshold, switch to the markers mode
+                  if (mapView.mapboxMap.cameraState.zoom < LABEL_THRESHOLD &&
+                      zoomState.value >= LABEL_THRESHOLD) {
+                    mapMode.value = MapMode.MARKERS
+                  }
+                  // On zoomin in past the threshold, switch to the user's selected mode
+                  if (mapView.mapboxMap.cameraState.zoom >= LABEL_THRESHOLD &&
+                      zoomState.value < LABEL_THRESHOLD) {
+                    mapMode.value = userMapMode.value
+                  }
+
+                  // store the zoom level
+                  // This must stay at the end of the listener.
+                  zoomState.value = mapView.mapboxMap.cameraState.zoom
                 }
+            // =======================  CAMERA  LISTENER  =======================
 
             onDispose {
-              pointAnnotationManager?.deleteAll()
+              markerAnnotationManager?.deleteAll()
```

Comment:
here shouldn't we delete all annotations of all kind? (rectangle and pLabel)

Reply:
Fixed